### PR TITLE
Add development docs for the extension

### PR DESCRIPTION
**Pull request by priyamsahoo**
The PR adds detailed documentation for developing the extension. It contains details about:
1. the project structure
2. extension code structure
3. language server code structure
4. testing style and structure for the project
5. developing and debugging the extension
6. developing webviews in the extension
7. additional resources

### CODE REVIEW

File: docs/development/project_structure.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+The project is divided into two main parts:
+
+1. The vscode-ansible extension (client)
+2. The ansible-language-server (server)
+
+Each part has its own package.json files, tests, and tsconfigs. This modularity
+helps in code navigation and development.
+
+!!! tip
```
Comment (alisonlhart) [inline docs/development/project_structure.md:11]:
```suggestion
[!TIP]
```
This is a [new markdown feature](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) that might be cool to include, optional nitpick.

File: docs/development/test_code.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+1. **unit-tests:** Runs the unit tests for the extension.
+2. **test-e2e:** Runs the end-to-end tests for the extension.
+3. **test-ui-current:** Runs the UI tests on the latest version of VS Code.
+4. **test-ui-oldest:** Runs the UI tests on the oldest supported version of VS
+   Code.
+
+!!! note
+
+    For the following tests (UI tests for the extension), you need to package and build a `.vsix file` of the extension by running `yarn run package`. As you run the UI tests, the .vsix file is automatically installed and the UI elements are checked.
+
```
Comment (alisonlhart) [inline docs/development/test_code.md:51]: Could add a note about MOCHA_GREP for UI tests, optional.

File: docs/development/webview_guide.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    - extension context file:
+      [createAnsibleProjectPage.ts](https://github.com/ansible/vscode-ansible/blob/main/src/features/contentCreator/createAnsibleProjectPage.ts)
+
+    - webview context file:
+      [createAnsibleProjectPageApp.ts](https://github.com/ansible/vscode-ansible/blob/main/src/webview/apps/contentCreator/createAnsibleProjectPageApp.ts)
+
+    - webpack config file:
+      [webpack.config.ts](https://github.com/ansible/vscode-ansible/blob/main/webpack.config.ts)
+
+2.  **External resources:** The best explanation of the implementing a webview
```
Comment (alisonlhart) [inline docs/development/webview_guide.md:214]:
```suggestion
2.  **External resources:** The best explanation of implementing a webview
```

File: docs/development/project_structure.md (hunk 1 above)
Comment (priyamsahoo) [reply docs/development/project_structure.md:11]: I was following the syntax for [mkdocs admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/). But I can give this a try.

Comment (priyamsahoo) [reply docs/development/project_structure.md:11]: I tested it out, and it does not render well in readthedocs. I will stick to the previous style.

Comment (priyamsahoo) [reply docs/development/project_structure.md:11]: This is how it looks on the documentation website:
Screenshot 2024-07-29 at 8 12 42 PM

File: docs/development/test_code.md (hunk 2 above)
Comment (priyamsahoo) [reply docs/development/test_code.md:51]: Could you give me more info about it that I can add?

File: docs/development/project_structure.md (hunk 1 above)
Comment (alisonlhart) [reply docs/development/project_structure.md:11]: Looks good!

File: docs/development/test_code.md (hunk 2 above)
Comment (alisonlhart) [reply docs/development/test_code.md:51]: You can call specific UI tests with the envvar MOCHA_GREP, like so:
`MOCHA_GREP="Test One Click Trial feature" yarn coverage-ui-with-mock-lightspeed-server` or 
`MOCHA_GREP="Verify the execution of playbook using ansible-navigator command" yarn coverage-ui-current` so you don't have to run all the tests in that set. I believe this is only for UI tests that use Mocha, but Tami could confirm more. I also tagged you in the slack thread where he explained it to me for more details.

Comment (alisonlhart) [reply docs/development/test_code.md:62]: Changes LGTM!

File: docs/development/webview_guide.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+### Step 5: Build and debug
+
+Build the extension using the command `yarn run webpack-dev` and run the
+extension in debug mode using the configuration `Launch extension`. Make changes
+in the code and reload the extension host window to see the live implementation.
+
+## Debugging the Webviews
+
+VS Code is built using built with Electron and includes Chrome's built-in
```
Comment (ariordan-redhat) [inline docs/development/webview_guide.md:105]:
```suggestion
VS Code is built with Electron and includes Chrome's built-in
```

File: docs/development/webview_guide.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+browser and DevTools. This setup allows debugging both VS Code and Webviews. You
+can follow these steps to debug a Webview in VS Code:
```
Comment (ariordan-redhat) [inline docs/development/webview_guide.md:107]:
```suggestion
browser and DevTools. This setup allows debugging both VS Code and Webviews.
Follow the steps below to debug a Webview in VS Code:
```

File: docs/development/webview_guide.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+   5 of Implementation steps).
+
+2. Navigate to and open the Webview you wish to debug within the editor.
+
+3. Press `Ctrl + Shift + P` (Windows/Linux) or `Cmd + Shift + P` (Mac) to bring
+   up the Command Palette.
+
+4. Search for the command `Open Webview Developer Tools` and select it.
+
+Once these steps are completed, an instance of DevTools will open in a separate
```
Comment (ariordan-redhat) [inline docs/development/webview_guide.md:119]:
```suggestion
After these steps are completed, an instance of DevTools will open in a separate
```

File: docs/development/webview_guide.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+window. You can use this instance to:
```
Comment (ariordan-redhat) [inline docs/development/webview_guide.md:120]:
```suggestion
window. You can use this instance to do the following:
```

File: docs/development/webview_guide.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+  },
+  undefined,
+  context.subscriptions,
+);
+```
+
+## Resources
+
+1.  **Local resources:** The extension has already implemented a few webviews
+    and you can take a look at the at the following places:
```
Comment (ariordan-redhat) [inline docs/development/webview_guide.md:203]:
```suggestion
1.  **Local resources:** The extension has already implemented some webviews.
    You can see them in the following locations:
```

**Review by ariordan-redhat — APPROVED** [review]
LGTM

**Pull request merged**